### PR TITLE
feat(modal): adjust close button alignment

### LINKED ISSUE #6964: Bug - Modal close button seems too low
`--pf-v6-c-modal-box__close--InsetBlockStart` should be a md spacer

image

### PR DESCRIPTION
**Pull request by srambach**
Fixes #6964 

Backstop report: https://drive.google.com/file/d/1uDWk76QpkLyzWxknMYCe0ckIlyI9necb/view?usp=sharing

### CODE REVIEW

File: src/patternfly/components/ModalBox/modal-box.scss [hunk 1]
```diff
@@ -61,7 +61,7 @@
   --#{$modal-box}__header--body--PaddingBlockStart: var(--pf-t--global--spacer--sm);
 
   // Close
-  --#{$modal-box}__close--InsetBlockStart: var(--#{$modal-box}__header--PaddingBlockStart);
+  --#{$modal-box}__close--InsetBlockStart: var(--pf-t--global--spacer--md);
```
Comment (mcoker) [inline src/patternfly/components/ModalBox/modal-box.scss:64]: WDYT about just offsetting the close button by its own padding? It's the same spacer as `var(--pf-t--global--spacer--md)` (16px/1rem) but will still be aligned if either the modal header's padding or button padding changes.

```suggestion
  --#{$modal-box}__close--InsetBlockStart: calc(var(--#{$modal-box}__header--PaddingBlockStart) - var(--pf-t--global--spacer--control--vertical--default));
```

**Review by mcoker — COMMENTED** [review]
LGTM, just one nit question

**Review by mcoker — APPROVED** [review]
LGTM! The only little nit I found could be adding in the `.pf-m-no-offset` modifier like we discussed and using it on this example that has a huge modal title - https://patternfly-pr-6970.surge.sh/components/modal/html/custom-title/

FWIW large title text is a use case for `.pf-m-no-offset` on card actions, too, so I think that would be consistent.

[modal-backstop.pdf](https://github.com/user-attachments/files/16618962/modal-backstop.pdf)

**Review by lboehling — APPROVED** [review]
very nice, thank you!!

**Comment by srambach** [comment]
Backstop https://drive.google.com/file/d/1VOinnKLy-MeRM66KdMOJqdhP82Fz3aEx/view?usp=sharing

File: src/patternfly/components/ModalBox/modal-box-header.hbs [hunk 2]
```diff
@@ -1,6 +1,11 @@
...
   {{#if modal-box-header--attribute}}
     {{{modal-box-header--attribute}}}
   {{/if}}>
-  {{> @partial-block}}
+  {{#> modal-box-header-main}}
+    {{> @partial-block}}
+  {{/modal-box-header-main}}
+    {{#if modal-box-header--HasHelp}}
+    {{> modal-box-header-help}}
+{{/if}}
```
Comment (mcoker) [inline src/patternfly/components/ModalBox/modal-box-header.hbs:10]: 😅

```suggestion
  {{#if modal-box-header--HasHelp}}
    {{> modal-box-header-help}}
  {{/if}}
```

File: src/patternfly/components/ModalBox/modal-box.scss [hunk 3]
```diff
@@ -37,6 +37,7 @@
 
   // Header main
   --#{$modal-box}__header-main--Gap: var(--pf-t--global--spacer--md);
+  --#{$modal-box}__header-main--PaddingBlockStart: var(--pf-t--global--spacer--sm);
```
Comment (mcoker) [inline src/patternfly/components/ModalBox/modal-box.scss:40]: wdyt about using `--pf-t--global--spacer--control--vertical--default` here? 

```suggestion
  --#{$modal-box}__header-main--PaddingBlockStart: var(--pf-t--global--spacer--control--vertical--default);
```

**Review by mcoker — COMMENTED** [review]
LGTM! I rebased locally and ran visual regressions and confirmed the modal is just 8px taller. Though I'm not sure why the standalone examples didn't indicate a change above the modal, while the demos with a modal in a page did, but I think it's a backstop thing.

[modal-8-19.24.pdf](https://github.com/user-attachments/files/16667383/modal-8-19.24.pdf)


Left a couple of comments - if you agree with the token change from `spacer--sm` to `spacer--control--vertical--default`, that would be good to get in before merging.

**Pull request merged**